### PR TITLE
feat: Added `expand` flag to `stringify_unsupported()` to expand series

### PR DESCRIPTION
**Pull request by SiddhantSadangi**
## Before submitting checklist

`stringify_unsupported()` does not currently expand lists, casting them to string instead.

For dicts that contain lists as values, as shown below
```
nested_dict = {
    "name": "John Doe",
    "age": 30,
    "is_employee": True,
    "address": {"street": "123 Main St", "city": "Anytown", "postal_code": "12345"},
    "projects": [
        {"name": "Project Alpha", "status": "completed", "budget": 100000},
        {"name": "Project Beta", "status": "in progress", "budget": 150000},
    ],
    "skills": ["Python", "Data Analysis", "Machine Learning"],
    "employment_history": [
        {"company": "Company A", "position": "Data Scientist", "years": 3},
        {"company": "Company B", "position": "Software Engineer", "years": 2},
    ],
}
```

this leads to keys like `projects`, `skills`, and `employment_history` being logged as a string
image
[Example](https://app.neptune.ai/siddhant.sadangi/sandbox/runs/details?viewId=standard-view&detailsTab=metadata&shortId=SAN1-107&path=dict%2F&attribute=employment_history)

This makes it harder to find relevant information if it is nested as a list in a dict, and also makes it a lot harder to download such metadata from Neptune, as it would be downloaded as a string.

This PR adds an optional `expand` flag to `stringify_unsupported()` to expand series like lists, sets, and tuples thereby storing nested values in their proper folder structure within enumerated folders, making it easier to both search for metadata and download it.
image
[Example](https://app.neptune.ai/siddhant.sadangi/sandbox/runs/details?viewId=standard-view&detailsTab=metadata&shortId=SAN1-115&type=run&path=dict%2Fprojects%2F0%2F&attribute=budget)

- [X] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [X] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/utils.py
Comment (kgodlewski) [inline src/neptune/utils.py:57]: I think a more in-depth explanation is necessary. Perhaps an additional example? cc @normandy7

**Review by kgodlewski — COMMENTED** [review]
I think we just need a better explanation, apart from that LGTM

**Review by kgodlewski — APPROVED** [review]
Let's wait for @normandy7's approval then it's good to merge

File: CHANGELOG.md
Comment (normandy7) [inline CHANGELOG.md:4]:
```suggestion
- Added optional `expand` flag to `stringify_unsupported()` to expand series ([#1862](https://github.com/neptune-ai/neptune-client/pull/1862))
```

File: src/neptune/utils.py
Comment (normandy7) [inline src/neptune/utils.py:70]:
```suggestion
        >>> run["complex_dict_expanded"] = stringify_unsupported(complex_dict, expand=True)
```

File: src/neptune/utils.py
Comment (normandy7) [inline src/neptune/utils.py:58]:
```suggestion
        expand (bool, optional): If True, the function expands series to store each item as an
        enumerated key-value pair. Otherwise, the entire series is logged as a string. Defaults to False.
```

**Review by normandy7 — CHANGES_REQUESTED** [review]

**Review by kgodlewski — APPROVED** [review]

**Review by normandy7 — APPROVED** [review]

**Pull request merged**
